src/prefect/events/schemas/automations.py

```diff
@@ -429,4 +432,112 @@ class AutomationCore(PrefectBaseModel, extra="ignore"):
 
 
 class Automation(AutomationCore):
-    id: UUID = Field(..., description="The ID of this automation")
+    id: Optional[UUID] = Field(default=None, description="The ID of this automation")
+
+    @classmethod
+    @sync_compatible
+    async def create(cls: Type[Self], automation: Type[Self]) -> UUID:
+        """
+        Create a new automation.
+
+        auto_to_create = Automation(
+        name="woodchonk",
+        trigger=EventTrigger(
+            expect={"animal.walked"},
+            match={
+                "genus": "Marmota",
+                "species": "monax",
+            },
+            posture="Reactive",
+            threshold=3,
+            within=timedelta(seconds=10),
+        ),
+        actions=[CancelFlowRun()]
+        )
+        created_automation = Automation.create(auto_to_create)
+        """
+        client, _ = get_or_create_client()
+        automation = AutomationCore(**automation.dict(exclude={"id"}))
+        automation = await client.create_automation(automation=automation)
+        return automation if automation else None
+
+    @classmethod
+    @sync_compatible
+    async def update(
+        cls: Type[Self], automation_id: UUID, automation: Type[Self]
+    ) -> UUID:
+        """
+        Updates an existing automation.
+
+        auto_to_update = Automation(
+        name="woodchonk",
+        trigger=EventTrigger(
+            expect={"animal.walked"},
+            match={
+                "genus": "Marmota",
+                "species": "monax",
+            },
+            posture="Reactive",
+            threshold=3,
+            within=timedelta(seconds=10),
+        ),
+        actions=[CancelFlowRun()]
+        )
+        updated_automation = Automation.update_create
+            automation_id="b3514963-02b1-47a5-93d1-6eeb131041cb",
+            automation=auto_to_update
+            )
+        """
+        client, _ = get_or_create_client()
+        automation = await client.update_automation(
+            automation_id=automation_id, automation=automation
+        )
+        return automation if automation else None
+
+    @classmethod
+    @sync_compatible
+    async def read(cls: Type[Self], id_or_name: str) -> Self:
```

Comment:
i think these should be separate kwargs, both initialized to `None`, with a `ValueError` if both are provided or if neither is provided